ts/components/ui/LogoPaymentExtended.tsx

```diff
@@ -6,9 +6,17 @@ import BpayLogo from "../../../img/wallet/payment-methods/bpay_logo_full.svg";
 import { BankLogoOrSkeleton } from "./utils/components/BankLogoOrLoadingSkeleton";
 export type LogoPaymentExtendedProps = {
   dimensions: { height: number; width: number };
-  icon?: "payPal" | "bpay";
-  abiCode?: string;
-};
+} & (
+  | {
+      abiCode: undefined;
+      icon: "payPal" | "bpay";
+    }
+  | {
+      icon: undefined;
+      abiCode: string | undefined;
+      imageA11yLabel?: string;
+    }
+);
```

Comment:
```suggestion
} & (
  | {
      icon: "payPal" | "bpay";
    }
  | {
      icon?: undefined;
      abiCode: string | undefined;
      imageA11yLabel?: string;
    }
);
```

Reply:
this way one of the two cases will suggest the addition of the other one, but TS will complain if you do.
after a bunch of research and trial and error I sincerely have no idea how this could be handled, maybe in some future evolution a higher number of args will ease the development, or nobody will care and it'll be good as-is 🤷 

nevertheless, I'm pushing these changes, I basically followed your suggestion but instead of `undefined` i used `never` for icon.